### PR TITLE
Document release process, prepare pyproject.toml for release

### PR DESCRIPTION
**Pull request by sritchie**
This PR:

- Adds the release process to DEVELOPING.md
- spruces up pyproject.toml, fixing errors from deploy and adding new optional keys
- adds a Quickstart the README.md

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -133,3 +133,60 @@ to render the notebooks into the static site directory.
...
+
+- Visit https://github.com/probcomp/genjax/releases/new to create a new release.
+- From the "Choose a tag" dropdown, type the new version (using the format
+  `v<MAJOR>.<MINOR>.<INCREMENTAL>`, like `v.0.1.0`) and select "Create new tag
+  on publish"
+- Fill out an appropriate title, and add release notes generated by looking at
+  PRs merged since the last release
+- Click "Publish Release"
+
+This will build and publish the new version to PyPI.
```
Comment (sritchie) [inline DEVELOPING.md:161]: This doesn't work yet, I need to add an action in the next PR. But this is configured from the pypi side.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <br>
 <p align="center">
-<img width="400px" src="./docs/assets/img/logo.png"/>
+<img width="400px" src="https://raw.githubusercontent.com/probcomp/genjax/main/docs/assets/img/logo.png?token=GHSAT0AAAAAABYGVMC6WSATH2PKWLSTATBIZMYON5A"/>
```
Comment (sritchie) [inline README.md:3]: This is so that the image will show up on the PyPI page.

File: README.md [hunk 3]
```diff
@@ -46,6 +46,77 @@ GenJAX is an implementation of Gen on top of [JAX](https://github.com/google/jax
...
+
+The following code snippet defines a generative function called `beta_bernoulli` that
+
+- takes a shape parameter `beta`
+- uses this to create and draw a value `p` from a [Beta
+  distribution](https://en.wikipedia.org/wiki/Beta_distribution)
+- Flips a coin that returns 1 with probability `p`, 0 with probability `1-p` and
+  returns that value
+
+JIT-compiles the function with JAX and then runs it with GenJAX:
```
Comment (sritchie) [inline README.md:82]: @femtomc give me some notes on how to make this exposition better, or maybe tell me I should delete it and just have the snippet now for Hello World purposes.

File: pyproject.toml [hunk 4]
```diff
@@ -1,16 +1,34 @@
...
+  "Sam Ritchie <sam@mentat.org>",
 ]
 license = "Apache 2.0"
 readme = "README.md"
 homepage = "https://github.com/probcomp/genjax"
 repository = "https://github.com/probcomp/genjax"
 documentation = "https://probcomp.github.io/genjax/"
-classifiers = ["Development Status :: 4 - Beta"]
+keywords = [
+  "artificial-intelligence",
```
Comment (sritchie) [inline pyproject.toml:20]: these came from the GitHub page.

**Review by sritchie — COMMENTED** [review]
Some notes...

File: README.md (hunk 2 above)
Comment (littleredcomputer) [reply README.md:3]: cool! the token is annoying though. Maybe there could be a public probcomp assets repo for branding some time in the future.

**Review by littleredcomputer — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (sritchie) [reply README.md:3]: Good call... once we make the repo public this issue goes away too.

File: README.md (hunk 3 above)
Comment (femtomc) [reply README.md:82]: I think we're going to have to work on a nice hello world example.

This one is pretty garbage, but it will suffice for now.

**Pull request merged**